src/main/perl/Lock.pm

```diff
@@ -230,6 +236,8 @@ is returned.
 sub unlock {
   my $self=shift;
   if ($self->{LOCK_SET}) {
+    flock ($self->{LOCK_FH}, LOCK_UN|LOCK_NB) or $self->error("cannot release flock on lock file: ",$self->{'LOCK_FILE'});
+    $self->{LOCK_FH}=undef;
```

Comment:
It would be preferable to `$self->{LOCK_FH}->close()` instead of `$self->{LOCK_FH}=undef`, as the intent is more explicit.  You should also test return status from the `close()` and report an error if you cannot close the lock file.